django/thunderstore/frontend/templatetags/auth_redirect.py

```diff
@@ -0,0 +1,9 @@
+from django import template
+from django.utils.safestring import mark_safe
+
+register = template.Library()
+
+
+@register.simple_tag(takes_context=True)
+def add_auth_redirect(context):
+    return mark_safe("?next=%s" % context.request.build_absolute_uri())
```

Comment:
This should be urlencoded appropriately. In fact, I'd say you might want to implement a custom social auth redirect which return s afull URL instead of conjoining the strings in the template renderer, otherwise you can't really handle URL encoding properly

Reply:
Not 100% sure if its done "right" right now, but I guess that's what you meant?